sql-cli/sql_cli/sql_directory_parser.py

```diff
@@ -135,15 +136,18 @@ def to_transform_operator(self) -> TransformOperator:
 
         :return: a TransformOperator
         """
-        return TransformOperator(
-            conn_id=self.metadata.get("conn_id"),
-            parameters=None,
-            handler=None,
-            database=self.metadata.get("database"),
-            schema=self.metadata.get("schema"),
-            python_callable=lambda: (str(self.path), None),
-            sql=self.content,
-        )
+        kwargs = {
+            "conn_id": self.metadata.get("conn_id"),
+            "parameters": None,
+            "handler": None,
+            "database": self.metadata.get("database"),
+            "schema": self.metadata.get("schema"),
+            "python_callable": lambda: (str(self.path), None),
+            "sql": self.content,
+        }
+        if airflow.__version__.startswith("2.2"):
```

Comment:
For dealing with versions we should use `packaging.version` module:

[python-sdk example](https://github.com/astronomer/astro-sdk/blob/main/python-sdk/tests/test_example_dags.py#L51-L55)

Reply:
@feluelle how do you think the packaging version could help in this specific line? I changed to use in the comparison, in the other part of the code, but it is not clear to be on the benefit in this particular case.

Reply:
I would say mainly for input validation.

Reply:
But the comment was more generally speaking. So it might make sense here to either use the specific string via `==` or the `Version()`. I know that `startswith` will probably work most of the time (except for `>2.20.x` 😅).